### PR TITLE
CI: Split downstream and upstream packages

### PR DESCRIPTION
**Pull request by rhaschke**
Address https://github.com/ros-planning/moveit/pull/2650#issuecomment-833646732

### CODE REVIEW

**Comment by rhaschke** [comment]
I have no clue why libXmu.so is missing now that I separated upstream and downstream packages. Need to finish for today.
`make[2]: *** No rule to make target '/usr/lib/x86_64-linux-gnu/libXmu.so', needed by '/home/runner/work/moveit/moveit/.work/target_ws/devel/.private/moveit_ros_perception/lib/libmoveit_mesh_filter.so.1.1.3'.  Stop.`

**Comment by rhaschke** [comment]
Invalidating the cache solved the build issue. Now, we have a failing pilz_planner test: #2653.
I guess, there are some files not installed in `moveit_resources` (because it worked before with a common workspace).

**Review by v4hn — COMMENTED** [review]
Looks good to me.

I don't trust my abilities to judge the details of the caching changes though, so I'll leave this to @tylerjw (who announced he has a long response time this month).

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -26,16 +27,17 @@ jobs:
...
+      # Pull any updates to the upstream workspace (after restoring it from cache)
       AFTER_SETUP_UPSTREAM_WORKSPACE: vcs pull $BASEDIR/upstream_ws/src
+      AFTER_SETUP_DOWNSTREAM_WORKSPACE: vcs pull $BASEDIR/downstream_ws/src
       AFTER_RUN_TARGET_TEST: ${{ matrix.env.CCOV && './.ci.prepare_codecov' || '' }}
       TARGET_CMAKE_ARGS: >
         -DCMAKE_BUILD_TYPE=${{ matrix.env.CCOV && 'RelWithDebInfo' || 'Release'}}
         ${{ matrix.env.CCOV && '-DCMAKE_CXX_FLAGS="--coverage" --no-warn-unused-cli' || '' }}
       CCACHE_DIR: "${{ github.workspace }}/.ccache"
       BASEDIR: ${{ github.workspace }}/.work
-      CACHE_PREFIX: "${{ matrix.env.IMAGE }}${{ matrix.env.CCOV && '-ccov' || '' }}"
```
Comment (tylerjw) [inline .github/workflows/ci.yaml:38]: Did you mean to remove this variable?

Comment (rhaschke) [reply .github/workflows/ci.yaml:38]: Yes, it's implemented locally for each caching step now.

**Comment by rhaschke** [comment]
The unit test fails due to a circular dependency issue: https://github.com/ros-planning/moveit/issues/2653#issuecomment-834307253

**Comment by v4hn** [comment]
In case anyone else wondered: you moved `moveit_resources` back to the target workspace to avoid the [circular dependency](https://github.com/ros-planning/moveit/pull/2654#issuecomment-839077917) mentioned above.

**Pull request merged**